rpmio/rpmkeyring.cc

```diff
@@ -61,6 +68,53 @@ rpmKeyring rpmKeyringFree(rpmKeyring keyring)
     return NULL;
 }
 
+rpmKeyringIterator rpmKeyringInitIterator(rpmKeyring keyring, int unused)
+{
+    if (!keyring || unused != 0)
+	return NULL;
+
+    return new rpmKeyringIterator_s {
+	rpmKeyringLink(keyring),
+	new rdlock(keyring->mutex),
+	keyring->keys.cbegin(),
+	NULL,
+    };
+}
+
+rpmPubkey rpmKeyringIteratorNext(rpmKeyringIterator iterator)
+{
+    rpmPubkey next = NULL;
+
+    if (!iterator)
+	return NULL;
+
+    while (iterator->iterator != iterator->keyring->keys.end()) {
+	next = iterator->iterator->second;
+	iterator->iterator++;
+	rdlock lock(next->mutex);
```

Comment:
I don't think we need this lock - the primary key of something in the keyring is not going to change after creation, no matter what :thinking:

Reply:
Yeah, that was may be a bit too conservative.